### ISSUE TITLE
Camel-quartz fails on Quarkus because of missing (optional) camel-cron dependency

### ISSUE THREAD

**Issue by nicolaferraro** [apache/camel-k-runtime#346]
I'm running the following route on camel-k with Quarkus JVM runtime:

```
from('quartz:trigger?cron=0/1+*+*+*+*+?')
    .routeId('groovy')
    .setHeader("m").constant("string!")
    .setBody()
        .simple('Magic${header.m}')
    .to('log:info?showAll=false')
```

The integration fails with:
```
Exception in thread "main" java.lang.ExceptionInInitializerError
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
        at java.base/java.lang.Class.newInstance(Class.java:584)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:60)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:38)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:106)
        at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:29)
Caused by: java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl.<clinit>(ApplicationImpl.zig:428)
        ... 9 more
Caused by: java.lang.NoClassDefFoundError: org/apache/camel/component/cron/api/CamelCronService
        at java.base/java.lang.ClassLoader.defineClass1(Native Method)
        at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1017)
        at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:174)
        at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:800)
        at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(BuiltinClassLoader.java:698)
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:621)
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:579)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
        at java.base/java.lang.Class.forName0(Native Method)
        at java.base/java.lang.Class.forName(Class.java:398)
        at io.quarkus.deployment.steps.Core$factoryFinderResolver19.deploy_0(Core$factoryFinderResolver19.zig:254)
        at io.quarkus.deployment.steps.Core$factoryFinderResolver19.deploy(Core$factoryFinderResolver19.zig:36)
        at io.quarkus.runner.ApplicationImpl.<clinit>(ApplicationImpl.zig:371)
        ... 9 more
Caused by: java.lang.ClassNotFoundException: org.apache.camel.component.cron.api.CamelCronService
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
        ... 23 more

```

The camel-quartz component in Camel has **optional** dependency on **camel-cron**, but while this works on a plain runtime, it fails on Quarkus (JVM).

**Comment by jamesnetherton** [comment]
Sounds like https://github.com/apache/camel-quarkus/issues/1195. It'll be fixed in camel-quarkus 1.0.0-CR1.

**Comment by nicolaferraro** [comment]
Cool, we can close this one.
